portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,12 +39,14 @@
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String fName = getParameter(request, "first-name", "");
```

Comment:
Best practice is to not hard-code strings in the code. 

You can define them as static constants at the top of the class. E.g. something like:

```java
private static final String FIRST_NAME_PARAM = "first-name"; 
...
```

Then you can use this constant wherever the string needs to be used.

Context: Hard-coded strings are referred to as Magic Strings. See this [StackOverflow](https://softwareengineering.stackexchange.com/questions/365339/what-is-wrong-with-magic-strings) question for more context. Same idea applies to numeric values. Its better to use define them as constants.